tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -1627,22 +1626,29 @@ async def inlay_hint(ls: ChapelLanguageServer, params: InlayHintParams):
     async def document_highlight(
         ls: ChapelLanguageServer, params: DocumentHighlightParams
     ):
-        text_doc = ls.workspace.get_text_document(params.text_document.uri)
+        text_doc_uri = params.text_document.uri
+        text_doc = ls.workspace.get_text_document(text_doc_uri)
 
         fi, _ = ls.get_file_info(text_doc.uri)
 
         node_and_loc = fi.get_use_or_def_segment_at_position(params.position)
         if not node_and_loc:
             return None
 
+        # only add highlights for the current document
+        should_add = lambda x: x.get_uri() == text_doc_uri
+
         # todo: it would be nice if this differentiated between read and write
-        highlights = [
-            DocumentHighlight(node_and_loc.rng, DocumentHighlightKind.Text)
+        highlights = []
+        if should_add(node_and_loc):
+            dh = DocumentHighlight(node_and_loc.rng, DocumentHighlightKind.Text)
+            highlights.append(dh)
+        uses = fi.uses_here.get(node_and_loc.node.unique_id(), [])
```

Comment:
`uses_here` should be a variable in this file info, right? Do you need `should_add` in the iterator?

Reply:
No it does not, I have simplified the code here by removing `should_add` from the iterator and inlining it in the if statement